AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2487,7 +2488,8 @@ - (void)__setSupernode:(ASDisplayNode *)newSupernode
       }
       
       // Now that we have a supernode, propagate its traits to self.
-      ASEnvironmentStatePropagateDown(self, [newSupernode environmentTraitCollection]);
+      ASPrimitiveTraitCollectionPropagateDown(self, newSupernode.primitiveTraitCollection);
```

Comment:
Let's call this `ASTraitCollectionPropagateDown`. In cases where there won't be ambiguity, we should omit the word `Primitive` because it's non-essential.